addon.py

```
@@ -418,6 +439,190 @@ def get_viewport_screenshot(self, max_size=800, filepath=None, format="png"):
+            scene.render.resolution_y = max_size
+            scene.render.filepath = filepath
+            scene.render.image_settings.file_format = format.upper()
+            bpy.ops.render.render(write_still=True)
+
+            if os.path.exists(filepath):
+                img = bpy.data.images.load(filepath)
+                width, height = img.size
+                bpy.data.images.remove(img)
+            else:
+                render_error = "Render file was not created"
+        except Exception as e:
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)